pkg/init/init.go

```diff
@@ -163,6 +167,10 @@ func (o *InitClient) downloadDirect(URL string, dest string) error {
 // downloadFromRegistry downloads a devfile from the provided registry and saves it in dest
 // If registryName is empty, will try to download the devfile from the list of registries in preferences
 func (o *InitClient) downloadFromRegistry(ctx context.Context, registryName string, devfile string, dest string) error {
+	// setting NewIndexSchema ensures that the Devfile library pulls registry based on the stack version
+	registryOptions := segment.GetRegistryOptions(ctx)
+	registryOptions.NewIndexSchema = true
```

Comment:
We have to check if we need a fallback if the new index schema is not supported by the Devfile Registry. Let's keep it for another PR. The issue is tracked with https://github.com/redhat-developer/odo/issues/6435